src/utils/logout.ts

```diff
@@ -2,23 +2,28 @@ import { logoutAction } from 'redux/commonActions';
 import { providerSelector } from 'redux/selectors/networkConfigSelectors';
 import { store } from 'redux/store';
 import { getIsLoggedIn } from 'utils/getIsLoggedIn';
+import { getProviderType } from 'utils/provider';
 
 export async function logout(
   callbackUrl?: string,
   onRedirect?: (callbackUrl?: string) => void
 ) {
   const provider = providerSelector(store.getState());
   const isLoggedIn = getIsLoggedIn();
+
   if (!isLoggedIn) {
     return;
   }
+
   store.dispatch(logoutAction());
+
   try {
     await provider.logout({ callbackUrl });
+
     if (callbackUrl) {
       if (onRedirect) {
         onRedirect(callbackUrl);
-      } else {
+      } else if (getProviderType(provider) !== 'wallet') {
```

Comment:
don't use strings, we have a function getIsProviderEqualTo(LoginMethodsEnum.wallet)